tests/integration/thrift_tests.py

```diff
@@ -406,6 +434,70 @@ def example(self, context):
 
         assert handler.edge_context == FakeEdgeContextFactory.DECODED_CONTEXT
 
+    def test_budget_header_pool_timeout(self):
+        """Test that the budget header is set in the headers with the pool timeout."""
+        budget = 100.0
+
+        class Handler(TestService.Iface):
+            def example(self, context):
+                self.context = context
+                return True
+
+        handler = Handler()
+
+        span_observer = mock.Mock(spec=SpanObserver)
+        with serve_thrift(handler, TestService) as server:
+            with baseplate_thrift_client(server.endpoint, TestService, span_observer) as context:
+                with context.example_service.retrying(attempts=3, budget=budget) as svc:
+                    svc.example()
+
+        assert handler.context.headers.get(b"Deadline-Budget").decode() == str(1000)
+
+    def test_budget_header_retry_timeout(self):
+        """Test that the budget header is set in the headers with the retry timeout."""
+        budget = 100.0
```

Comment:
🔕 this is really nitpicky but maybe use `0.1` instead of `100.0` so we are actually testing 100ms instead of 100s (which would be too unrealistic compare to real code). You would also need to change the assert on line 475 to `str(int(budget *1000))` instead.